tests/unit/test_json_schema_validation_rule_functions.py

```diff
@@ -0,0 +1,355 @@
+    ValidationRuleName,
+    filter_unused_inputted_rules,
+    check_for_duplicate_inputted_rules,
+    check_for_conflicting_inputted_rules,
+    get_rule_from_inputted_rules,
+    get_js_type_from_inputted_rules,
+    get_in_range_parameters_from_inputted_rule,
+    get_regex_parameters_from_inputted_rule,
+    _get_parameters_from_inputted_rule,
+    _get_names_from_inputted_rules,
+    _get_name_from_inputted_rule,
+    _get_rules_by_name,
+)
+from schematic.schemas.constants import JSONSchemaType
+
+
+@pytest.mark.parametrize(
+    "input_list, expected_list",
+    [
+        ([], []),
+        (["str"], ["str"]),
+        (["str", "unused_rule"], ["str"]),
+    ],
+    ids=["No rules", "Str rule", "Contains an unused rule"],
+)
+def test_filter_unused_inputted_rules(
+    input_list: list[str], expected_list: list[str]
+) -> None:
+    """
+    Test for filter_unused_inputted_rules
+    Tests that only rules used to create JSON Schemas are left
+    """
+    result = filter_unused_inputted_rules(input_list)
+    assert result == expected_list
+
+
+@pytest.mark.parametrize(
+    "input_list",
+    [
+        ([]),
+        (["str"]),
+        (["str", "int", "bool"]),
+    ],
+    ids=["No rules", "One rule", "Multiple rules"],
+)
+def test_check_for_duplicate_inputted_rules(
+    input_list: list[str],
+) -> None:
+    """
+    Test for check_for_duplicate_inputted_rules
+    Tests that no duplicate rules were found and no exceptions raised
+    """
+    check_for_duplicate_inputted_rules(input_list)
+
+
+@pytest.mark.parametrize(
+    "input_list",
+    [(["str", "str"]), (["str warning", "str error"])],
+    ids=["Duplicate string rules", "Duplicate string rules with different parameters"],
+)
+def test_check_for_duplicate_inputted_rules_with_duplicates(
+    input_list: list[str],
+) -> None:
+    """
+    Test for check_for_duplicate_inputted_rules
+    Tests that duplicate rules were found and a ValueError was raised
+    """
+    with pytest.raises(ValueError, match="Validation Rules contains duplicates"):
+        check_for_duplicate_inputted_rules(input_list)
+
+
+@pytest.mark.parametrize(
+    "input_list",
+    [([]), (["str", "regex"]), (["inRange", "int"])],
+    ids=["No rules", "Str and regex rules", "InRange and int rules"],
+)
+def test_check_for_conflicting_inputted_rules(
+    input_list: list[str],
+) -> None:
+    """
+    Test for check_for_conflicting_inputted_rules
+    Tests that no rules are in conflict with each other
+    """
+    check_for_conflicting_inputted_rules(input_list)
+
+
+@pytest.mark.parametrize(
+    "input_list",
+    [
+        (["str", "int"]),
+        (["date", "url"]),
+        (["inRange", "str"]),
+        (["regex", "int"]),
+    ],
+    ids=[
+        "Multiple type rules",
+        "Multiple format rules",
+        "InRange and str rules",
+        "Regex and int rules",
+    ],
+)
+def test_check_for_conflicting_inputted_rules_with_conflicts(
```

Comment:
the test cases here are fine. I am thinking if one rule has multiple incompatible rules, can those incompatible rules all be surfaced?